nixos/modules/security/krb5/krb5-conf-format.nix

```diff
@@ -0,0 +1,87 @@
+{ pkgs, lib, ... }:
+
+# Based on
+# - https://web.mit.edu/kerberos/krb5-1.12/doc/admin/conf_files/krb5_conf.html
+# - https://manpages.debian.org/unstable/heimdal-docs/krb5.conf.5heimdal.en.html
+
+let
+  inherit (lib) boolToString concatMapStringsSep concatStringsSep filter
+    isAttrs isBool isList mapAttrsToList mdDoc mkOption;
+  inherit (lib.types) attrsOf bool either int listOf oneOf path str submodule;
+in
+{ }: {
+  type = let
+    section = attrsOf relation;
+    relation = either (attrsOf value) value;
+    value = either (listOf atom) atom;
+    atom = oneOf [int str bool];
+  in submodule {
+    freeformType = attrsOf section;
+    options = {
+      include = mkOption {
+        default = [ ];
+        description = mdDoc ''
+          Files to include in the Kerberos configuration.
+        '';
+        type = listOf path;
+      };
```

Comment:
Usually I would use plural too but here it's to keep consistency in naming with options you can specify multiple times which are also singular:

```
kdc = [
  "kerberos.mit.edu"
  "kerberos-1.mit.edu"
  "kerberos-2.mit.edu:750"
];
```

Using `coercedTo` so you can specify a single one as is seems like a good idea though. Should I also use that in the definition of the normal config option as opposed to `value = either (listOf atom) atom;`?

Reply:
I think the `coercedTo` is just a shorthand to avoid having to do any postprocessing in config, if you know the postprocessing is only going to be type coercion anyway, or if it greatly simplifies the config block. I don't think it's necessary for the freeform stuff, since we're already doing a bunch of other things in the generator. It might actually be more readable and clear the way it is.